hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/volume/HddsVolume.java

```diff
@@ -257,14 +258,23 @@ public synchronized VolumeCheckResult check(@Nullable Boolean unused)
     VolumeCheckResult result = super.check(unused);
 
     DatanodeConfiguration df = getConf().getObject(DatanodeConfiguration.class);
+    if (isDbLoadFailure()) {
+      LOG.warn("Volume {} failed to access RocksDB: RocksDB parent directory is null, " +
+          "the volume might not have been loaded properly.", getStorageDir());
+      return VolumeCheckResult.FAILED;
+    }
     if (result != VolumeCheckResult.HEALTHY ||
         !df.getContainerSchemaV3Enabled() || !isDbLoaded()) {
       return result;
     }
 
     // Check that per-volume RocksDB is present.
-    File dbFile = new File(dbParentDir, CONTAINER_DB_NAME);
-    if (!dbFile.exists() || !dbFile.canRead()) {
+    File dbFile = dbParentDir == null ? null : new File(dbParentDir, CONTAINER_DB_NAME);
```

Comment:
If it passes L261 check, does it still have the chance to have a null dbParentDir?

Reply:
`dbParentDir` will non-null if it passes L261 check. I think this can be simplified